runtime/moonbeam/src/precompiles.rs

```diff
@@ -57,38 +60,52 @@ where
 {
 	fn execute(
+		&self,
 		address: H160,
 		input: &[u8],
 		target_gas: Option<u64>,
 		context: &Context,
-	) -> Option<Result<PrecompileOutput, ExitError>> {
+		is_static: bool,
+	) -> Option<PrecompileResult> {
 		match address {
 			// Ethereum precompiles :
-			a if a == hash(1) => Some(ECRecover::execute(input, target_gas, context)),
-			a if a == hash(2) => Some(Sha256::execute(input, target_gas, context)),
-			a if a == hash(3) => Some(Ripemd160::execute(input, target_gas, context)),
-			a if a == hash(4) => Some(Identity::execute(input, target_gas, context)),
-			a if a == hash(5) => Some(Modexp::execute(input, target_gas, context)),
-			a if a == hash(6) => Some(Bn128Add::execute(input, target_gas, context)),
-			a if a == hash(7) => Some(Bn128Mul::execute(input, target_gas, context)),
-			a if a == hash(8) => Some(Bn128Pairing::execute(input, target_gas, context)),
-			a if a == hash(9) => Some(Blake2F::execute(input, target_gas, context)),
+			a if a == hash(1) => Some(ECRecover::execute(input, target_gas, context, is_static)),
+			a if a == hash(2) => Some(Sha256::execute(input, target_gas, context, is_static)),
+			a if a == hash(3) => Some(Ripemd160::execute(input, target_gas, context, is_static)),
+			a if a == hash(4) => Some(Identity::execute(input, target_gas, context, is_static)),
+			a if a == hash(5) => Some(Modexp::execute(input, target_gas, context, is_static)),
+			a if a == hash(6) => Some(Bn128Add::execute(input, target_gas, context, is_static)),
+			a if a == hash(7) => Some(Bn128Mul::execute(input, target_gas, context, is_static)),
+			a if a == hash(8) => Some(Bn128Pairing::execute(input, target_gas, context, is_static)),
+			a if a == hash(9) => Some(Blake2F::execute(input, target_gas, context, is_static)),
 			// Non-Moonbeam specific nor Ethereum precompiles :
-			a if a == hash(1024) => Some(Sha3FIPS256::execute(input, target_gas, context)),
-			a if a == hash(1025) => Some(Dispatch::<R>::execute(input, target_gas, context)),
-			a if a == hash(1026) => Some(ECRecoverPublicKey::execute(input, target_gas, context)),
+			a if a == hash(1024) => {
+				Some(Sha3FIPS256::execute(input, target_gas, context, is_static))
+			}
+			a if a == hash(1025) => Some(Dispatch::<R>::execute(
+				input, target_gas, context, is_static,
+			)),
+			a if a == hash(1026) => Some(ECRecoverPublicKey::execute(
+				input, target_gas, context, is_static,
+			)),
 			// Moonbeam specific precompiles :
 			a if a == hash(2048) => Some(ParachainStakingWrapper::<R>::execute(
-				input, target_gas, context,
+				input, target_gas, context, is_static,
 			)),
 			a if a == hash(2049) => Some(CrowdloanRewardsWrapper::<R>::execute(
-				input, target_gas, context,
+				input, target_gas, context, is_static,
 			)),
 			_ => None,
 		}
 	}
+	fn is_precompile(&self, address: H160) -> bool {
+		Self::used_addresses()
+			.collect::<sp_std::vec::Vec<R::AccountId>>()
+			.contains(&R::AddressMapping::into_account_id(address))
```

Comment:
```suggestion
			.find(|x| x == &R::AddressMapping::into_account_id(address))
			.is_some()
```
Avoids collecting/cloning.